processing/src/main/java/io/druid/query/aggregation/hyperloglog/PreComputedHyperUniquesSerde.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.aggregation.hyperloglog;
+
+import io.druid.data.input.InputRow;
+import io.druid.hll.HyperLogLogCollector;
+import io.druid.hll.HyperLogLogHash;
+import io.druid.segment.serde.ComplexMetricExtractor;
+import org.apache.commons.codec.binary.Base64;
+
+import java.nio.ByteBuffer;
+import java.util.List;
+
+public class PreComputedHyperUniquesSerde extends HyperUniquesSerde
+{
+  public PreComputedHyperUniquesSerde(HyperLogLogHash hyperLogLogHash)
+  {
+    super(hyperLogLogHash);
+  }
+
+  @Override
+  public ComplexMetricExtractor getExtractor()
+  {
+    return new ComplexMetricExtractor()
+    {
+      @Override
+      public Class<HyperLogLogCollector> extractedClass()
+      {
+        return HyperLogLogCollector.class;
+      }
+
+      @Override
+      public HyperLogLogCollector extractValue(InputRow inputRow, String metricName)
+      {
+        Object rawValue = inputRow.getRaw(metricName);
+
+        if (rawValue instanceof HyperLogLogCollector) {
+          return (HyperLogLogCollector) rawValue;
+        } else {
+          HyperLogLogCollector collector = HyperLogLogCollector.makeLatestCollector();
+
+          List<String> dimValues = inputRow.getDimension(metricName);
```

Comment:
I'd suggest doing something similar to SketchMergeComplexMetricSerde here.

- use `inputRow.getRaw(metricName)` to get the actual value from the input row, not transformed into a `List<String>`
- handle some types it could be; see `SketchHolder.deserialize` for an example. For this one I'd suggest handling String, byte[], and HyperLogLogCollector.

This should be faster and should also work better with input rows that come from non-text sources, where the field might be a byte array or might already be a HLLC object. It'd also explicitly _not_ handle lists of HLLs which I think is not something that really makes sense, or that we'd want to support anyway.